src/mo_frontend/typing.ml

```diff
@@ -1150,7 +1150,18 @@ and infer_exp'' env exp : T.typ =
         { env with async = C.NullCap; in_actor = true }
       else env
     in
-    infer_obj env' obj_sort.it dec_fields exp.at
+    let t = infer_obj env' obj_sort.it dec_fields exp.at in
+    begin match typ_opt with
+      | Some typ ->
+        let t' = check_typ env' typ in
```

Comment:
I am not super confident with this `exp'`, but tried to model this code as close to the `class` check as possible.

Reply:
Yes, I'm not sure what's already in that env' to be honest. For example, does it contain the actor identifier (probably), or  private types defined within the body (hopefully not)?